File: flutter-lib/lib/src/impl/decorations_align.dart
enum DecorationsAlign {
  hleft,
  hright,
  vleft,
  vright;

  static DecorationsAlign fromString(String? s) {
    switch (s?.trim().toLowerCase()) {
      case 'hright':
        return hright;
      case 'vleft':
        return vleft;
      case 'vright':
        return vright;
      default:
        return hleft;
    }
  }

  static DecorationsAlign? fromJson(String? s) => s != null ? fromString(s) : null;
  static String? toJson(DecorationsAlign? a) => a?.name;

  bool get isVertical => this == vleft || this == vright;
  bool get isAtStart => this == vleft || this == hleft;
}

File: swt_native/src/main/java/dev/equo/swt/DecorationsAlign.java
package dev.equo.swt;

public enum DecorationsAlign {
    HLEFT, HRIGHT, VLEFT, VRIGHT;

    public static DecorationsAlign fromString(String s) {
        if (s == null) return HLEFT;
        switch (s.trim().toLowerCase()) {
            case "hright": return HRIGHT;
            case "vleft":  return VLEFT;
            case "vright": return VRIGHT;
            default:       return HLEFT;
        }
    }

    public boolean isVertical() { return this == VLEFT || this == VRIGHT; }
    public boolean isAtStart()  { return this == VLEFT || this == HLEFT; }
}
